### ISSUE TITLE
Collinear points break line stroking.

### ISSUE THREAD

**Issue by AdyanOchirov** [vancluever/z2d#37]
Doing `path.lineTo` on collinear points makes `context.stroke` panic on `@intFromFloat` because of infinities.
``` zig
pub fn main() !void {
    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
    const allocator = gpa.allocator();

    const surface = try z2d.Surface.init(.image_surface_rgba, allocator, 200, 200);
    defer surface.deinit();

    var context: z2d.Context = .{
        .surface = surface,
        .anti_aliasing_mode = .none, //or default, both panic
    };
    var path = z2d.Path.init(allocator);
    defer path.deinit();

    try path.moveTo(10, 10);
    try path.lineTo(20, 20);
    try path.lineTo(30, 30);
    try path.lineTo(40, 10);
    try path.close(); //also panics without this

    try context.stroke(allocator, path); //panics here
}
```

**Comment by vancluever** [comment]
Thanks @AdyanOchirov! I've got this set up in a repro now and am investigating.

There's no handling of colinear joins yet so this is not necessarily surprising to me, and should be hopefully easy to fix.

**Comment by AdyanOchirov** [comment]
@vancluever It still breaks with going-back collinear points.
```
try path.moveTo(10, 10);
try path.lineTo(30, 30);
try path.lineTo(20, 20);
try path.lineTo(40, 10);
```

**Comment by vancluever** [comment]
@AdyanOchirov just wanted to let you know I haven't forgotten about this - I've been working on trying to set up so that we can handle these cases in a much simpler way! This is next up so I'm going to re-open this.

**Comment by vancluever** [comment]
@AdyanOchirov incidentally, looks like c690797 fixed it! I'm going to add the examples to the colinear test to make sure there's no regressions before I work on simplifying some of the intersection code, and am going to close this off in the meantime. Let me know if you find any other cases!

**Comment by vancluever** [comment]
Added in c78a523.

**Comment by vancluever** [comment]
Re-opening as I think I thought of another case that needs fixing due to a regression that might result in twisting when the previously known direction is clockwise and a co-linear join happens. Will check in the morning.

**Comment by vancluever** [comment]
Fixed again in 155619d.
